core/src/main/java/jenkins/model/ProjectNamingStrategy.java

```diff
@@ -99,6 +101,11 @@ public DefaultProjectNamingStrategy() {
         @Override
         public void checkName(String origName) throws Failure {
             // default - should just do nothing (this is how Jenkins worked before introducing this ExtensionPoint)
+            if (StringUtils.isNotBlank(DEFAULT_PATTERN) && StringUtils.isNotBlank(origName)) {
```

Comment:
and afair this strategy can be changed in global config, shouldn't strict check be done in some other routine i.e. `ProjectNamingStrategy.checkName() and call from implementations to super.checkname()`?